src/components/ViewCourse/sections/AddReserve.js

```diff
@@ -9,7 +9,7 @@ class AddReserve extends React.Component {
   static contextType = CalloutContext; // Too Much Magic
 
   static propTypes = {
-    courseListingId: PropTypes.string.isRequired,
+    courseListingId: PropTypes.string,
```

Comment:
Why this change? I can't think of a legitimate scenario where `<AddReserve>` is called without there being a course-listing.

Reply:
Just to remove console errors. The alternative is to pass in an empty string or have a defaultProp. Given how vocal prop-types is when missing required props, I've always thought of `isRequired` as "we need this otherwise we'll straight-up crash"

Reply:
So what is the scenario when this could happen?

Reply:
When the page is loading. The form renders everything, but the course info hasn't been fetched from the backend yet, so the `courseListingId` is still unknown.

Reply:
Ah, of course! Yes, thank you. That characteristic issue of React-based sites.